Source/WebCore/editing/TextIterator.cpp

```diff
@@ -623,9 +623,20 @@ void TextIterator::handleTextRun()
         unsigned textRunStart = m_textRun->start();
         unsigned runStart = std::max(textRunStart, start);
 
+        unsigned textRunEnd = textRunStart + m_textRun->length();
+        unsigned runEnd = std::min(textRunEnd, end);
+
+        // Determine what the next text run will be, but don't advance yet
+        auto nextTextRun = InlineIterator::nextTextBoxInLogicalOrder(m_textRun, m_textRunLogicalOrderCache);
+
         // Check for collapsed space at the start of this run.
         bool needSpace = m_lastTextNodeEndedWithCollapsedSpace || (m_textRun == firstTextRun && textRunStart == runStart && runStart);
         if (needSpace && !renderer.style().isCollapsibleWhiteSpace(m_lastCharacter) && m_lastCharacter) {
+            if (runStart >= runEnd && m_behaviors.contains(TextIteratorBehavior::IgnoresWhiteSpaceAtEndOfRun)) {
+                m_textRun = nextTextRun;
```

Comment:
What if the next text run was also entirely consisting of whitespace?

Reply:
I'm not sure I understand your question 😅

The idea behind this code is to do similar to what's done on :688 in that if we've reached the end of the desired range (i.e. `runStart >= runEnd`) then we should stop processing things and instead just keep iterating until we run out of text runs to iterate.

When I first read this code it just struck me as odd that we didn't check `runStart < runEnd` for the collapsed whitespace case.  My gut is telling me that this should always be the behavior in all cases (i.e. we shouldn't be reading text beyond the end of the range we want), but I created `TextIteratorBehavior::IgnoresWhiteSpaceAtEndOfRun` to minimize any risk of other issues (since this is a fix for a crash, i wouldn't want the change to have to get rolled out because it caused some other problem).

Reply:
i added some tests for this (assuming i understood your point 😅)